docs/MockFunctionAPI.md

```diff
@@ -313,146 +492,76 @@ Syntactic sugar function for:
 ```
 
-Example usage:
+Useful together with `.mockResolvedValueOnce()` or to reject with different exceptions over multiple async calls:
+
+<Tabs groupId="examples">
+<TabItem value="js" label="JavaScript">
 
 ```js
 test('async test', async () => {
   const asyncMock = jest
     .fn()
     .mockResolvedValueOnce('first call')
-    .mockRejectedValueOnce(new Error('Async error'));
+    .mockRejectedValueOnce(new Error('Async error message'));
 
-  await asyncMock(); // first call
-  await asyncMock(); // throws "Async error"
+  await asyncMock(); // 'first call'
+  await asyncMock(); // throws 'Async error message'
 });
 ```
 
-## TypeScript
-
-Jest itself is written in [TypeScript](https://www.typescriptlang.org).
-
-If you are using [Create React App](https://create-react-app.dev) then the [TypeScript template](https://create-react-app.dev/docs/adding-typescript/) has everything you need to start writing tests in TypeScript.
-
-Otherwise, please see our [Getting Started](GettingStarted.md#using-typescript) guide for to get setup with TypeScript.
-
-You can see an example of using Jest with TypeScript in our [GitHub repository](https://github.com/facebook/jest/tree/main/examples/typescript).
+</TabItem>
 
-### `jest.MockedFunction`
+<TabItem value="ts" label="TypeScript">
 
-> `jest.MockedFunction` is available in the `@types/jest` module from version `24.9.0`.
+```ts
+test('async test', async () => {
+  const asyncMock = jest
+    .fn<() => Promise<string>>()
+    .mockResolvedValueOnce('first call')
+    .mockRejectedValueOnce(new Error('Async error message'));
 
-The following examples will assume you have an understanding of how [Jest mock functions work with JavaScript](MockFunctions.md).
+  await asyncMock(); // 'first call'
+  await asyncMock(); // throws 'Async error message'
+});
+```
 
-You can use `jest.MockedFunction` to represent a function that has been replaced by a Jest mock.
+</TabItem>
+</Tabs>
 
-Example using [automatic `jest.mock`](JestObjectAPI.md#jestmockmodulename-factory-options):
+## TypeScript Usage
 
-```ts
-// Assume `add` is imported and used within `calculate`.
-import add from './add';
-import calculate from './calc';
-
-jest.mock('./add');
+If you are using [Create React App](https://create-react-app.dev) then the [TypeScript template](https://create-react-app.dev/docs/adding-typescript/) has everything you need to start writing tests in TypeScript.
 
-// Our mock of `add` is now fully typed
-const mockAdd = add as jest.MockedFunction<typeof add>;
+Otherwise, please see our [Getting Started](GettingStarted.md#using-typescript) guide for to get setup with TypeScript.
 
-test('calculate calls add', () => {
-  calculate('Add', 1, 2);
+You can see an example of using Jest with TypeScript in our [GitHub repository](https://github.com/facebook/jest/tree/main/examples/typescript).
 
-  expect(mockAdd).toBeCalledTimes(1);
-  expect(mockAdd).toBeCalledWith(1, 2);
-});
-```
+### `jest.fn(implementation?)`
```

Comment:
By the way, I double checked if the example works. All is correct. Will add it to `examples` folder later. TypeScript example will need more changes.

Reply:
convert back to draft so I don't accidentally merge? 😀